vividus-extension-visual-testing/src/main/java/org/vividus/visual/steps/AbstractVisualSteps.java

```diff
@@ -43,11 +43,15 @@ protected AbstractVisualSteps(IUiContext uiContext, IAttachmentPublisher attachm
         this.softAssert = softAssert;
     }
 
-    protected <T extends VisualCheck> VisualCheckResult execute(Function<T, VisualCheckResult> checkResultProvider,
+    protected <T extends VisualCheck> Optional<VisualCheckResult> execute(
```

Comment:
hmm, according to your changes, the result of this method is not used, so the better option would be to change option so that it returns `void`

ps: sorry for confusion, I've not checked the code before putting my comment about `Optional`